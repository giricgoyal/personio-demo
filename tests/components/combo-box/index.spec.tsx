import { fireEvent } from '@testing-library/react'
import React from 'react'
import ComboBox from 'src/components/combo-input'
import { render } from 'tests/test-utils'

describe('components/combo-box', () => {
    let container, getByTestId, onChangeHandler
    const dataList = ['John Doe', 'Jane Austin', 'Mark Roger']

    beforeEach(() => {
        onChangeHandler = jest.fn()
        ;({ container, getByTestId } = render(
            <ComboBox id="combo-box" onChange={onChangeHandler} dataList={dataList} />,
        ))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('On value change', () => {
        beforeEach(() => {
            fireEvent.change(getByTestId('input'), {
                target: {
                    value: 'john',
                },
            })
        })

        test('should render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })

        test('should call the onChangeHandler', () => {
            expect(onChangeHandler).toHaveBeenCalledWith('john')
        })
    })
})
