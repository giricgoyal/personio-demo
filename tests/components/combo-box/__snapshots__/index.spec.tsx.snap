// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/combo-box On value change should render and match snapshot 1`] = `
<div>
  <input
    class="combo-box"
    data-testid="input"
    list="combo-box"
  />
  <datalist
    id="combo-box"
  >
    <option
      value="John Doe"
    />
    <option
      value="Jane Austin"
    />
    <option
      value="Mark Roger"
    />
  </datalist>
</div>
`;

exports[`components/combo-box should render and match snapshot 1`] = `
<div>
  <input
    class="combo-box"
    data-testid="input"
    list="combo-box"
  />
  <datalist
    id="combo-box"
  >
    <option
      value="John Doe"
    />
    <option
      value="Jane Austin"
    />
    <option
      value="Mark Roger"
    />
  </datalist>
</div>
`;
